### PR TITLE
A script to verify the SHA1 checksums of binaries

### PR DESCRIPTION
**Pull request by everett-toews**

### CODE REVIEW

File: script/verify [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  echo "Example: $0 https://abc-123.ssl.cf5.rackcdn.com/releases/test_1.1/"
+  exit 1
+fi
+
+set -euo pipefail
+IFS=$'\n\t'
+
+BASE_URL=$1
+
+files=(
```
Comment (ycombinator) [inline script/verify]: Could this list be generated from the files under the `bin` folder (generated earlier in the build process by `script/cross`)?

**Comment by ycombinator** [comment]
Currently, the release process (as codified in `script/release`) is:
* Cross-compile the binaries (`script/cross`),
* Generate the checksums (`script/checksum`), and
* Publish the binaries and checksum files (`script/publish`)

Looking at this script, I imagine the new release process would be:
* Cross-compile the binaries (`script/cross`),
* Generate the checksums (`script/checksum`),
* Publish the binaries and checksum files (`script/publish`) *to the test release "folder" in the CloudFiles container*,
* *Verify the checksums (`script/verify-sha1`), and
* *Re-publish* the binaries and checksum files (`script/publish`) to the regular release "folder" in the CloudFiles container.

Is that what you had in mind, @everett-toews?

**Comment by everett-toews** [comment]
@ycombinator Yep. That's what I had in mind. 

The re-publish step would ideally be done as a Cloud Files [copy operation](http://docs.rackspace.com/files/api/v1/cf-devguide/content/COPY_copyobject_v1__account___container___object__objectServicesOperations_d1e000.html).

**Comment by ycombinator** [comment]
Cool. I think I'll redo `script/publish` to always publish to test, while also renaming it to something like `script/publish-to-test`. Then I'll create a new `script/publish-to-prod` that will do a CloudFiles COPY from the test folder.

File: script/verify (hunk 1 above)
Comment (everett-toews) [inline script/verify]: Good call. Fixed.

**Comment by ycombinator** [comment]
@everett-toews I'd like to use `verify-sha` programmatically inside `script/release` to determine whether to move to the next step in the release process (step #5 from my earlier comment) or not. As such, could you make `verify-sha1` fail fast upon a checksum mismatch and exit with a non-zero code?

**Comment by everett-toews** [comment]
@ycombinator Alright. I think this is okay now. Give it a go.

**Pull request merged**